src/main/java/unibook/logic/commands/EditCommand.java

```diff
@@ -57,6 +89,69 @@ public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editModuleDescriptor = null;
+    }
+
+    /**
+     * @param index of the person in the filtered module list to edit
+     * @param editModuleDescriptor details to edit the module with
+     */
+    public EditCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = index;
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+        this.editPersonDescriptor = null;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException, ModuleNotFoundException {
+        requireNonNull(model);
+
+        if (this.editModuleDescriptor == null) {
+            List<Person> lastShownList = model.getFilteredPersonList();
+            List<Module> latestModList = model.getFilteredModuleList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+
+            Person personToEdit = lastShownList.get(index.getZeroBased());
+            if (!editPersonDescriptor.getModules().equals(Optional.empty()) && latestModList.size() != 0) {
+                Module checkMod = editPersonDescriptor.getModules().get().iterator().next();
+                if (!latestModList.contains(checkMod)) {
+                    throw new ModuleNotFoundException();
+                }
+            }
+            Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+
+            if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+                throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            }
+
+            model.setPerson(personToEdit, editedPerson);
```

Comment:
same idea with the new person object created. bidirectionality **must** be maintained. If you don't update the person object pointed to in a list of modules that have that person as student/prof, then they will point to outdated old objects -> memory wastage, app logic wont work properly. 

Go through all the associated modules with the person you are editing, and change the person object pointed to in the student/prof list (whichever they are a part of) in each module to the newly created person object.